CommunicationLayer/MessagingFramework/BatteryMessage.cpp

```diff
@@ -0,0 +1,246 @@
+{
+    return getUnsignedInt(messageData_, VOLTAGE);
+}
+
+unsigned int BatteryMessage::current() const
+{
+    return getUnsignedInt(messageData_, CURRENT);
+}
+
+unsigned short BatteryMessage::fan0Speed() const
+{
+    return getUnsignedShort(messageData_, FAN_0_SPEED);
+}
+
+unsigned short BatteryMessage::fan1Speed() const
+{
+    return getUnsignedShort(messageData_, FAN_1_SPEED);
+}
+
+unsigned short BatteryMessage::fanContactors12VCurrentConsumption() const
+{
+    return getUnsignedShort(messageData_, FAN_AND_CONTACTORS_12V_CURRENT_CONSUMPTION);
+}
+
+unsigned short BatteryMessage::cmu12VCurrentConsumption() const
+{
+    return getUnsignedShort(messageData_, CMU_12V_CURRENT_CONSUMPTION);
+}
+
+bool BatteryMessage::bmsCanLockedOut() const
+{
+    return static_cast<bool>(messageData_.at(BMS_CAN_LOCKED_OUT));
+}
+
+
+QString BatteryMessage::toString() const
+{
+    // TODO chars don't need QString::number
+    QString messageString;
+    messageString += QString::number(MessageDefines::Battery) + ", ";
+    messageString += QString::number(alive()) + ", ";
+    messageString += QString::number(packSocAmpHours()) + ", ";
+    messageString += QString::number(packSocPercentage()) + ", ";
+    messageString += QString::number(packBalanceSoc()) + ", ";
+    messageString += QString::number(packBalanceSocPercentage()) + ", ";
+    messageString += QString::number(chargingCellVoltageError()) + ", ";
+    messageString += QString::number(cellTemperatureMargin()) + ", ";
+    messageString += QString::number(dischargingCellVoltageError()) + ", ";
+    messageString += QString::number(totalPackCapacity()) + ", ";
+    messageString += QString::number(prechargeContactorDriverStatus()) + ", ";
+    messageString += QString::number(prechargeState()) + ", ";
+    messageString += QString::number(prechargeTimerElapsed()) + ", ";
+    messageString += QString::number(prechargeTimerCount()) + ", ";
+    messageString += QString::number(lowestCellVoltage()) + ", ";
+    messageString += QString::number(lowestCellVoltageNumber()) + ", ";
+    messageString += QString::number(highestCellVoltage()) + ", ";
+    messageString += QString::number(highestCellVoltageNumber()) + ", ";
+    messageString += QString::number(lowestCellTemperature()) + ", ";
+    messageString += QString::number(lowestCellTemperatureNumber()) + ", ";
+    messageString += QString::number(highestCellTemperature()) + ", ";
+    messageString += QString::number(highestCellTemperatureNumber()) + ", ";
+    messageString += QString::number(voltage()) + ", ";
+    messageString += QString::number(current()) + ", ";
+    messageString += QString::number(fan0Speed()) + ", ";
+    messageString += QString::number(fan1Speed()) + ", ";
+    messageString += QString::number(fanContactors12VCurrentConsumption()) + ", ";
+    messageString += QString::number(cmu12VCurrentConsumption()) + ", ";
+    messageString += QString::number(bmsCanLockedOut()) + ", ";
```

Comment:
I have been more or less following the convention of the previous author, but I was wondering why we terminate the string with another comma? Is this because the messages are appended together? If so, should OtherMessage.cpp toString() not follow this convention and terminate without a comma?